SEED-platform/seed
# Add geocoding confidence to database

#### Any background context you want to provide?
Add a field to store the geocoding results into the database.

#### What's this PR do?
* Add `geocoding_confidence` to the database. Currently, this is a number but based on [this page](https://developer.mapquest.com/documentation/geocoding-api/quality-codes/), it appears that a string may be better. *Let me know if you want me to change this to a string.*
* Fixes tests related to the new column
* Minor updates to the documentation
* Use `django.contrib.gis.db.backends.postgis` engine in the other Django config targets.

#### How should this be manually tested?
* Import a new dataset
* Ensure that Geocoding Confidence doesn't appear as a mappable column in the mapping suggestions
* Map, match, pair data
* Go to inventory page and check if column exists in the list settings

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
![screen shot 2019-01-05 at 07 12 05](https://user-images.githubusercontent.com/1907354/50725527-fa8d3e00-10bb-11e9-94d0-8d6686c5ad05.png)

#### Other Notes


I'm getting a unit test error for some reason. Not sure what it is about. Error is below:

```
======================================================================
ERROR: test_geocode_address_can_handle_addresses_with_reserved_and_unsafe_characters (seed.tests.test_utils_geocode.GeocodeAddresses)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nlong/working/seed/seed/tests/test_utils_geocode.py", line 254, in test_geocode_address_can_handle_addresses_with_reserved_and_unsafe_characters
    geocode_addresses(properties)
  File "/Users/nlong/working/seed/seed/utils/geocode.py", line 51, in geocode_addresses
    address_geocodings = _address_geocodings(id_addresses)
  File "/Users/nlong/working/seed/seed/utils/geocode.py", line 120, in _address_geocodings
    response = requests.get(request_url)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/requests/sessions.py", line 524, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/raven/breadcrumbs.py", line 326, in send
    resp = real_send(self, request, *args, **kwargs)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/requests/sessions.py", line 637, in send
    r = adapter.send(request, **kwargs)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/urllib3/connectionpool.py", line 377, in _make_request
    httplib_response = conn.getresponse(buffering=True)
  File "/Users/nlong/.pyenv/versions/seed-3.6.5/lib/python3.6/site-packages/vcr/stubs/__init__.py", line 237, in getresponse
    self.cassette.record_mode)
vcr.errors.CannotOverwriteExistingCassetteException: No match for the request (<Request (GET) https://www.mapquestapi.com/geocoding/v1/batch?&inFormat=json&outFormat=json&thumbMaps=false&maxResults=1&json=%7B%22locations%22:%20[%7B%22street%22:%20%223001%20Brighton%20Blvd,%20suite%202693,%20Denver,%20Colorado,%2080216%22%7D]%7D&key=removed>) was found. Can't overwrite existing cassette ('seed/tests/data/vcr_cassettes/geocode_reserved_and_unsafe_characters.yaml') in your current record mode ('once').
```

Also, there appear to be two warnings about geocoding on the mapping screen:
![screen shot 2019-01-05 at 07 11 00](https://user-images.githubusercontent.com/1907354/50725545-37593500-10bc-11e9-98c7-b8f3e21777ec.png)



#### Definition of Done:
